src/density_methods.jl

```diff
@@ -143,20 +143,24 @@ end
 function atomic_density_superposition(basis::PlaneWaveBasis{T},
                                       form_factors::IdDict{Tuple{Int,T},T};
                                       coefficients=ones(T, length(basis.model.atoms))
-                                      )::Array{T,3} where {T}
+                                      ) where {T}
     model = basis.model
-    G_cart = G_vectors_cart(basis)
+    G_cart = to_cpu(G_vectors_cart(basis))
+    Gv = to_cpu(G_vectors(basis))
+    atom_groups = to_cpu(model.atom_groups)
+    positions = to_cpu(model.positions)
```

Comment:
These should not be on the GPU. Are you sure this is needed?

Reply:
Indeed, I have deleted those. The model is always on CPU.